service/src/BatchServer.cpp

```diff
@@ -78,7 +79,20 @@ namespace geopm
             this->create_shmem();
         };
         auto run = [this]() {
-            this->run_batch();
+            try {
+                this->run_batch();
+            }
+            catch(...) {
+                if (m_is_client_waiting) {
+                    std::cerr << "Warning: <geopm>: " << __FILE__ << ":" << __LINE__
+                              << " Batch server was terminated while client was waiting: sending client quit message\n";
```

Comment:
Minor note, it seems that the `BatchServer` may be invoked by the `geopmd` as just a server process, without any connection to the command line ... perhaps we could use the `syslog()` instead for writing error messages? I have used it before.

Reply:
This detail is managed by dasbus.